packages/report-server/src/__tests__/reportBuilder/transform/fillRows.test.ts

```diff
@@ -0,0 +1,282 @@
+        { period: '20200108', organisationUnit: 'TO', BCD1: 'filled' }, // preceding value in list is 20200102
+        { period: '20200103', organisationUnit: 'TO', BCD1: 3 },
+        { period: '20200105', organisationUnit: 'TO', BCD1: 5 },
+        { period: '20200104', organisationUnit: 'TO', BCD1: 'filled' }, // preceding value in list is 20200105
+        { period: '20200106', organisationUnit: 'TO', BCD1: 6 },
+      ]),
+    );
+  });
+
+  it('supports expressions in the column values', async () => {
+    const transform = buildTestTransform([
+      {
+        transform: 'fillRows',
+        requiredColumnValues: [
+          {
+            column: 'date',
+            values:
+              "= dateUtils.eachDayOfInterval({ start: date('2019-12-31'), end: date('2020-01-07')}).map(dateToString)",
+          },
+        ],
+        missingRowValues: {
+          BCD1: 'filled',
+          organisationUnit: 'TO',
+        },
+      },
+    ]);
+    expect(await transform(TransformTable.fromRows(FILLABLE_ANALYTICS_WITH_DATE))).toStrictEqual(
+      TransformTable.fromRows([
+        { date: '2019-12-31', organisationUnit: 'TO', BCD1: 'filled' },
+        { date: '2020-01-01', organisationUnit: 'TO', BCD1: 1 },
+        { date: '2020-01-02', organisationUnit: 'TO', BCD1: 2 },
+        { date: '2020-01-03', organisationUnit: 'TO', BCD1: 3 },
+        { date: '2020-01-04', organisationUnit: 'TO', BCD1: 'filled' },
+        { date: '2020-01-05', organisationUnit: 'TO', BCD1: 5 },
+        { date: '2020-01-06', organisationUnit: 'TO', BCD1: 6 },
+        { date: '2020-01-07', organisationUnit: 'TO', BCD1: 'filled' },
+      ]),
+    );
+  });
+
+  it('supports expressions in the missing rows values', async () => {
+    const transform = buildTestTransform([
+      {
+        transform: 'fillRows',
+        requiredColumnValues: [
+          {
+            column: 'period',
+            values: [
+              '20191231',
+              '20200101',
+              '20200102',
+              '20200103',
+              '20200104',
+              '20200105',
+              '20200106',
+              '20200107',
+            ],
+          },
+        ],
+        missingRowValues: {
+          BCD1: '= @index > 1 ? @current.BCD1 : 0',
+          organisationUnit: "= 'org unit ' + @index",
+        },
+      },
+    ]);
+    expect(await transform(TransformTable.fromRows(FILLABLE_ANALYTICS))).toStrictEqual(
+      TransformTable.fromRows([
+        { period: '20191231', organisationUnit: 'org unit 1', BCD1: 0 },
+        { period: '20200101', organisationUnit: 'TO', BCD1: 1 },
+        { period: '20200102', organisationUnit: 'TO', BCD1: 2 },
+        { period: '20200103', organisationUnit: 'TO', BCD1: 3 },
+        { period: '20200104', organisationUnit: 'org unit 3', BCD1: 3 }, // Indexing is a little awkward due to not knowing rows are being filled
```

Comment:
Don't love that the new index doesn't get determined dynamically. But it's consistent with `insertRows` and the lookups in the viz-builder don't get used too heavily and are in need of a re-write anyway.